javaparser-core-testing/src/test/java/com/github/javaparser/utils/SourceRootPathResolutionTest.java

```diff
@@ -1,56 +1,46 @@
 package com.github.javaparser.utils;
 
-import com.github.javaparser.StaticJavaParser;
-import com.github.javaparser.ast.CompilationUnit;
 import org.junit.jupiter.api.Test;
 
+import java.io.IOException;
+import java.nio.file.Files;
 import java.nio.file.Path;
 import java.nio.file.Paths;
-import java.util.Optional;
 
 import static org.junit.jupiter.api.Assertions.assertEquals;
 
 class SourceRootPathResolutionTest {
 
+    private static Path N(Path p) { return p.toAbsolutePath().normalize(); }
+
     @Test
-    void relativeKey_resolvesUnderNewRoot() {
-        Path newRoot = Paths.get("/out");
-        Path oldRoot = Paths.get("/proj");
-        Path key = Paths.get("src/main/java/p/A.java");
+    void relative_is_resolved_under_newRoot_currentBehaviour() throws IOException {
 
-        Path result = SourceRoot.resolveSavePath(newRoot, oldRoot, key, Optional.empty());
-        assertEquals(Paths.get("/out/src/main/java/p/A.java"), result);
-    }
+        Path srcRoot = Files.createTempDirectory("jp-src-");
```

Comment:
Once the path resolution method has been isolated, there is no longer any need to create test directories.